jenkins/Jenkinsfile.macos

```diff
@@ -0,0 +1,274 @@
+          }
+        }
+
+        when {
+          anyOf {
+            expression { return FLAVOR == 'Electron' }
+            expression { return FLAVOR == 'Desktop' && IS_PRO == true } // Only build Qt on Pro
+          }
+        }
+
+        stages {
+          stage('Sequential Matrix') {
+            options {
+              lock('synchronous-matrix')
+            }
+            stages{
+              stage('Build and Sign') {
+
+                environment {
+                  AWS_ACCOUNT_ID = '749683154838'
+                  KEYCHAIN_PASSPHRASE = credentials('ide-keychain-passphrase')
+                }
+
+                steps {
+                  script {
+                    ENV = utils.getBuildEnv(!params.DAILY)
+                  }
+                  // unlock keychain to ensure build gets signed.
+                  sh 'security unlock-keychain -p ${KEYCHAIN_PASSPHRASE} && security set-keychain-settings' // turn off timeout
+                  // build rstudio
+                  dir ("package/osx") {
+                    withAWS(role: 'build', roleAccount: AWS_ACCOUNT_ID) {
+                      sh "${ENV} ./make-package clean --rstudio-target=${FLAVOR}"
+                    }
+                  }
+                }
+              }
+
+              stage('Test') {
+                environment {
+                  PATH = "${env.HOME}/opt/bin:${env.PATH}"
+                }
+                steps {
+                  dir("package/osx/build/src/cpp") {
+                    // attempt to run cpp unit tests
+                    // problems with rsession finding openssl, so those tests
+                    // are disabled until we solve it (#6890)
+                    sh "arch -x86_64 ./rstudio-tests"
+                    
+                    // electron tests not working in CI environment, under investigation
+                    // if(FLAVOR == 'Electron') {
+                    //   sh "cd ../../../../../src/node/desktop/ && $HOME/.yarn/bin/yarn && $HOME/.yarn/bin/yarn test"
+                    // }
+                    }
+                  }
+                }
+
+              stage('Notarize and Upload') {
+                when { expression { return params.PUBLISH } }
+                
+                environment {
+                  PATH = "${env.HOME}/opt/bin:${env.PATH}"
+                  PACKAGE_FILE = """${sh (
+                    script: "basename `ls package/osx/build/RStudio-*.dmg`",
+                    returnStdout: true
+                  ).trim()}"""
+                  BUILD_TYPE = """${sh (
+                    script: "cat version/BUILDTYPE",
+                    returnStdout: true
+                  ).trim().toLowerCase()}"""
+                  PRODUCT = "${utils.getProductName()}"
+                }
+                
+                stages {
+
+                  stage("Notarize") {
+                    environment {
+                      APPLE_ID = credentials('ide-apple-notarizer')
```

Comment:
Is this change in specifying the credentials the reason why the `notarize-release.sh` changed?

Reply:
Yes - when you assign a `username:password` environment variable in this way you get `VAR_NAME=username:password`, `VAR_NAME_USR=username` and `VAR_NAME_PSW=password`. IIUC, it's more idiomatic for declarative pipeline syntax and I was trying to keep syntax consistent